2.0.11 fails to build on Rust 1.57.0

**Issue Description**
This currently fails to build. The abbreviated log is:

```
     Running `rustc --crate-name rnsslapd --edition=2018 librnsslapd/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type staticlib --crate-type lib --emit=dep-info,link -C opt-level=3 -C panic=abort -C debuginfo=2 -C metadata=783a645e755712d2 -C extra-filename=-783a645e755712d2 --out-dir /build/389-ds-base/src/389-ds-base-389-ds-base-2.0.11/rs/rnsslapd/release/deps -L dependency=/build/389-ds-base/src/389-ds-base-389-ds-base-2.0.11/rs/rnsslapd/release/deps --extern libc=/build/389-ds-base/src/389-ds-base-389-ds-base-2.0.11/rs/rnsslapd/release/deps/liblibc-16c7bac821acdeab.rlib --extern slapd=/build/389-ds-base/src/389-ds-base-389-ds-base-2.0.11/rs/rnsslapd/release/deps/libslapd-3724b8e7071b7d09.rlib`
error: unused return value of `CString::from_raw` that must be used
  --> librnsslapd/src/lib.rs:35:25
   |
35 |                         CString::from_raw(raw);
   |                         ^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> librnsslapd/src/lib.rs:1:9
   |
1  | #![deny(warnings)]
   |         ^^^^^^^^
   = note: `#[deny(unused_must_use)]` implied by `#[deny(warnings)]`
   = note: call `drop(from_raw(ptr))` if you intend to drop the `CString`

error: could not compile `librnsslapd` due to previous error

Caused by:
  process didn't exit successfully: `rustc --crate-name rnsslapd --edition=2018 librnsslapd/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type staticlib --crate-type lib --emit=dep-info,link -C opt-level=3 -C panic=abort -C debuginfo=2 -C metadata=783a645e755712d2 -C extra-filename=-783a645e755712d2 --out-dir /build/389-ds-base/src/389-ds-base-389-ds-base-2.0.11/rs/rnsslapd/release/deps -L dependency=/build/389-ds-base/src/389-ds-base-389-ds-base-2.0.11/rs/rnsslapd/release/deps --extern libc=/build/389-ds-base/src/389-ds-base-389-ds-base-2.0.11/rs/rnsslapd/release/deps/liblibc-16c7bac821acdeab.rlib --extern slapd=/build/389-ds-base/src/389-ds-base-389-ds-base-2.0.11/rs/rnsslapd/release/deps/libslapd-3724b8e7071b7d09.rlib` (exit status: 1)
make: *** [Makefile:12750: /build/389-ds-base/src/389-ds-base-389-ds-base-2.0.11/rs/rnsslapd/release/librnsslapd.a] Error 101
make: *** Waiting for unfinished jobs....
```

**Package Version and Platform:**
 - Platform: Arch Linux
 - Package and version: 2.0.11 git tag

**Steps to Reproduce**
Steps to reproduce the behavior: https://github.com/archlinux/svntogit-packages/blob/packages/389-ds-base/trunk/PKGBUILD#L39

**Expected results**
Well, I guess it should compile cleanly.

--- a/src/librnsslapd/src/lib.rs
+++ b/src/librnsslapd/src/lib.rs
@@ -7,7 +7,7 @@
 extern crate libc;
 use libc::c_char;
 use slapd;
-use std::ffi::{CStr, CString};
+use std::ffi::CStr;
 
 #[no_mangle]
 pub extern "C" fn do_nothing_again_rust() -> usize {
@@ -29,12 +29,7 @@ pub extern "C" fn fernet_generate_token(dn: *const c_char, raw_key: *const c_cha
                 Ok(tok) => {
                     // We have to move string memory ownership by copying so the system
                     // allocator has it.
-                    let raw = tok.into_raw();
-                    let dup_tok = unsafe { libc::strdup(raw) };
-                    unsafe {
-                        CString::from_raw(raw);
-                    };
-                    dup_tok
+                    unsafe { libc::strdup(tok.as_c_str().as_ptr()) }
                 }
                 Err(_) => std::ptr::null_mut(),
             }

